### PR TITLE
Enrich context with step output for code generation

### PR DESCRIPTION
**Pull request by alexandrudanpop**
Fixes OPS-2177

## Testing Checklist

Check all that apply:

- [x] I tested the feature thoroughly, including edge cases

- [x] I verified all affected areas still work as expected

- [x] Automated tests were added/updated if necessary

- [x] Changes are backwards compatible with any existing data, otherwise a migration script is provided

## Visual check
On UX

### CODE REVIEW

File: packages/shared/src/lib/ai/chat/code-output-structure.ts [hunk 1]
```diff
@@ -1,9 +1,25 @@
...
+});
+
+export type CodeSchema = z.infer<typeof codeLLMSchema>;
+
+export const unifiedCodeLLMSchema = z.object({
+  type: z.enum(['code', 'reply']),
+  code: z.string().optional().describe(codeDescription),
+  packageJson: z.string().optional().describe(packageJsonDescription),
+  textAnswer: z.string().describe(textAnswerDescription),
 });
```
Comment (alexandrudanpop) [inline packages/shared/src/lib/ai/chat/code-output-structure.ts:23]: Doing this because union and discriminated union types are not properly supported. We should recheck after the update to v5 of `ai-sdk`.

File: packages/server/api/src/app/ai/chat/ai-mcp-chat.controller.ts [hunk 2]
```diff
@@ -264,7 +264,9 @@ export const aiMCPChatController: FastifyPluginAsyncTypebox = async (app) => {
       const { aiConfig, languageModel } = llmConfigResult;
 
       const enrichedContext = request.body.additionalContext
-        ? await enrichContext(request.body.additionalContext, projectId)
+        ? await enrichContext(request.body.additionalContext, projectId, {
+            includeCurrentStepOutput: IncludeOptions.ALWAYS,
```
Comment (alexandrudanpop) [inline packages/server/api/src/app/ai/chat/ai-mcp-chat.controller.ts:269]: Always include the output for better context. It might be that the code output does not have the data the user needs, so it can be improved on next calls.

File: packages/server/api/src/app/ai/chat/ai-mcp-chat.controller.ts [hunk 3]
```diff
@@ -279,14 +281,20 @@ export const aiMCPChatController: FastifyPluginAsyncTypebox = async (app) => {
...
             ...messages,
             assistantMessage,
           ]);
         },
         onError: (error) => {
-          logger.error('Failed to generate code', error);
+          logger.error('Failed to generate code', {
+            error,
+          });
+          throw error;
```
Comment (alexandrudanpop) [inline packages/server/api/src/app/ai/chat/ai-mcp-chat.controller.ts:298]: when there are generation errors throw

File: packages/server/api/src/app/ai/chat/prompts.service.ts [hunk 4]
```diff
@@ -58,16 +58,28 @@ export const getBlockSystemPrompt = async (
...
+
+      if (enrichedContext?.steps?.some((s) => s.variables)) {
+        enhancedPrompt += `
+        \n\n ## Inputs properties and sample values:\n${JSON.stringify(
+          enrichedContext.steps.map((s) =>
+            s.variables?.map((v) => ({
+              [`inputs.${v.name}`]: v.value,
+            })),
+          ),
+        )}\n\n`;
```
Comment (alexandrudanpop) [inline packages/server/api/src/app/ai/chat/prompts.service.ts:75]: better describe inputs structure

File: .cursor/rules/coding-guidelines.mdc [hunk 5]
```diff
@@ -78,8 +101,10 @@ export function getUserProfile(userId: string): UserProfile {
 ```
 
 ## Frontend Guidelines
+
 - Use the `react-ui` package for main application logic.
 - Place pure, reusable components in the `ui-components` package, documented in Storybook.
+- Don't do breaking changes to existing code interfaces (like component props or names, unless this was discussed)
```
Comment (alexandrudanpop) [inline .cursor/rules/coding-guidelines.mdc:107]: just a small but important update :)

File: packages/react-ui/src/app/features/builder/ai-chat/step-settings-ai-conversation.tsx [hunk 6]
```diff
@@ -111,24 +108,29 @@ const getMessageId = (message: MessageType, idx: number): string => {
...
+              content: parsed.textAnswer,
+            },
+          ],
         },
-      ],
-    },
-  };
+      }
+    : createMessage(message, idx);
 };
```
Comment (cezudas) [inline packages/react-ui/src/app/features/builder/ai-chat/step-settings-ai-conversation.tsx]: Consider returning early instead of using a ternary expression here — it would improve readability and reduce nesting.

**Review by cezudas — APPROVED** [review]

File: packages/react-ui/src/app/features/builder/ai-chat/step-settings-ai-conversation.tsx (hunk 6 above)
Comment (alexandrudanpop) [reply packages/react-ui/src/app/features/builder/ai-chat/step-settings-ai-conversation.tsx]: fixed
